### DISCUSSION TITLE
API Rules (Row Level Access rules)

### DISCUSSION THREAD

**Discussion by DDuran19** [pocketbase/pocketbase#5705] [Q&A] [answered]
Good day! I have a "feature flag" on my application which is stored on features collection. The goal of the collection is to control the features that are live on my application. With that in mind, I created an an "announcements" collection which basically is a feature of my app that allows me as an admin to send notifications to them. 

I want to leverage the API rules of Pocketbase with this.  On my announcements collection, I only want someone to fetch data on it when the feature is turned on. I read on the docs that in order to reference a different collection, I should use `@collection`. so here's what I did

`@collection.features.name = "announcements" && @collection.features.status = "on"`

In my mind, it should look for the features collection whose name is "announcements" and the status is "on" 

Unfortunately, it doesn't work the way I think it would be. I know I am doing something wrong and there's something off with my API rule I set, so I was wondering if anyone can help me figure this out. Thanks so much !!

Here's what my features collection looks like:
![image](https://github.com/user-attachments/assets/548ae3c3-50b4-40fc-9c84-280c4fc18420)

**Comment by ganigeorgiev** [comment, marked as answer]
Filter fields that results in more than one row (`@collection.*`, multiple `relation`, back-relations, multiple `select`, etc.) apply by default a "match-all" condition, aka. requires every record/value from the set to satisfy the specific condition.

In your case you want to target one or more records from the "features" collection so in that case prefixing the operator with `?` should work since that will force "any/at-least-one" type of constraint, aka. try with:

```js
// require to have at least one features record with name "announcements" AND that has status "on"
//
// or as a more generic explanation - similar to an SQL WHERE clause, you could think of the rule as "funnel"
// where with each condition you are narrowing down the matching records
@collection.features.name ?= "announcements" && @collection.features.status ?= "on"
```

If that doesn't work, please export your collections configuration as json (_Admin UI > Settings > Export collections_).

**Reply by ganigeorgiev** [reply]
To further clarify, the above rule is expected to generate a pseudo SQL like:

```sql
SELECT announcements.*
FROM announcements
LEFT JOIN features
WHERE features.name = "announcements" AND features.status = "on"
...
```

**Comment by DDuran19** [comment]
Superb explanation, thanks man, you're the best for real!
